UPGRADE.rst

```diff
@@ -94,11 +94,13 @@ Requirement for X-Forwarded-Proto header
 When using Synapse with a reverse proxy (in particular, when using the
 `x_forwarded` option on an HTTP listener), Synapse now expects to receive an
 `X-Forwarded-Proto` header on incoming HTTP requests. If it is not set, Synapse
-will log a warning on each received request.
+will log a warning on each received request. Synapse also requires the `Host`
+header to be preserved.
 
 To avoid the warning, administrators using a reverse proxy should ensure that
 the reverse proxy sets `X-Forwarded-Proto` header to `https` or `http` to
-indicate the protocol used by the client. See the `reverse proxy documentation
+indicate the protocol used by the client and `Host` header matching
+`public_baseurl` in the config. See the `reverse proxy documentation
```

Comment:
it's actually important that the header is *preserved*, rather than that it matches the url in the config: the entire point is that synapse needs to know whether it needs to redirect the client to the correct host.

So I think it's a mistake to mention `public_baseurl` here; and in any case, the host header won't have any effect on the warning.